apps/linkedin-to-notion/src/background/index.ts

```diff
@@ -8,9 +8,12 @@ chrome.action.onClicked.addListener((tab) => {
   }
 });
 
-chrome.tabs.onUpdated.addListener((tabId, _, tab) => {
-  if (tab.url.match(linkedInURLRegex)) {
-    return chrome.tabs.sendMessage(tabId, 'updateLinkedInNotionSidePanel');
+chrome.tabs.onUpdated.addListener((tabId, { status }, tab) => {
+  if (tab.url?.match(linkedInURLRegex)) {
+    if (status === 'complete') {
+      chrome.tabs.sendMessage(tabId, 'updateLinkedInNotionSidePanel');
+    }
+  } else {
+    chrome.tabs.sendMessage(tabId, 'closeLinkedInNotionSidePanel');
   }
```

Comment:
~Still don't think `?` is needed~

~And not a big fan of the syntax. I'd rather use returns 😬~

Reply:
I'd do this:
```

chrome.tabs.onUpdated.addListener((tabId, { status }, tab) => {
  if (!tab.url.match(linkedInURLRegex)) {
    return chrome.tabs.sendMessage(tabId, 'closeLinkedInNotionSidePanel');
  }

  if (status === 'complete') {
    return chrome.tabs.sendMessage(tabId, 'updateLinkedInNotionSidePanel');
  }
});

```

Reply:
My bad ! I found why you needed ?

But go for `if (tab.url && tab.url.match(linkedInURLRegex)) {` instead

Reply:
Or in my example, the ? is just fine
```
chrome.tabs.onUpdated.addListener((tabId, { status }, tab) => {
  if (!tab.url?.match(linkedInURLRegex)) {
    return chrome.tabs.sendMessage(tabId, 'closeLinkedInNotionSidePanel');
  }

  if (status === 'complete') {
    return chrome.tabs.sendMessage(tabId, 'updateLinkedInNotionSidePanel');
  }
});
```

Reply:
The problem with your suggestion is that it will try and update the side panel on every linkedin page, even if it's not a profile page, as long as the update status is "complete". Correct me if I'm wrong but the whole iframe does not unmount when it's 'closed', it only switches its css class to 'hidden'.

What about:
```suggestion
  if (tab.url && tab.url.match(linkedInURLRegex)) {
    if (status === 'complete') {
      chrome.tabs.sendMessage(tabId, 'updateLinkedInNotionSidePanel');
    }
  } else {
    chrome.tabs.sendMessage(tabId, 'closeLinkedInNotionSidePanel');
  }
```

Reply:
Fair question! Go for your solution 👌